### PR TITLE
Chore: Update e2e selectors for Transformations

### LINKED ISSUE #78411: BI: Migrate from aria-label e2e selectors to data-testid
Hi @grafana/grafana-bi-squad!

A while back to changed to prefer using `data-testid` for E2E selectors instead of `aria-label`. There's only a few of these old `aria-label`-based selectors left, so we've made this list of files that your team owns so we can tidy them up 🙌.

For some guidance on how exactly to fix these, check out [this comment](https://github.com/grafana/grafana/issues/36523#issuecomment-1819160512).

There are 2 issues over 1 file:

- [ ] 2 issues in `public/app/features/transformers/editors/ReduceTransformerEditor.tsx`

### PR DESCRIPTION
**Pull request by baldm0mma**
**What is this feature?**

Update e2e selector from `aira-label` to `data-testid`.

**Why do we need this feature?**

Update to better testing practices.

**Who is this feature for?**

Everyone.

**Which issue(s) does this PR fix?**:

Fixes #78411

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-e2e-selectors/src/selectors/components.ts [hunk 1]
```diff
@@ -230,7 +230,7 @@ export const Components = {
     card: (name: string) => `data-testid New transform ${name}`,
     Reduce: {
       modeLabel: 'Transform mode label',
```
Comment (codeincarnate) [inline packages/grafana-e2e-selectors/src/selectors/components.ts:232]: Should this be changed too?

**Review by codeincarnate — COMMENTED** [review]
Maybe one issue?

File: packages/grafana-e2e-selectors/src/selectors/components.ts (hunk 1 above)
Comment (baldm0mma) [reply packages/grafana-e2e-selectors/src/selectors/components.ts:232]: The e2e tests aren't actually attempting to select this label, but I think that's a good idea. I'll also add a test to verify.

Comment (codeincarnate) [reply packages/grafana-e2e-selectors/src/selectors/components.ts:232]: Ah okay, looks good then 👍

**Review by codeincarnate — APPROVED** [review]

**Review by oscarkilhed — APPROVED** [review]

**Pull request merged**